pkg/apps/controller/deploymentconfig/deploymentconfig_controller.go

```diff
@@ -128,28 +128,44 @@ func (c *DeploymentConfigController) Handle(config *appsapi.DeploymentConfig) er
 			return err
 		}
 	}
+
+	// Never deploy with invalid or unresolved images
+	for i, container := range config.Spec.Template.Spec.Containers {
+		if strings.TrimSpace(container.Image) == "" {
+			glog.V(4).Infof("Postponing rollout #%d for DeploymentConfig %s/%s because of invalid or unresolved image for container #%d (name=%s)", config.Status.LatestVersion, config.Namespace, config.Name, i, container.Name)
+			return c.updateStatus(config, existingDeployments, true)
+		}
+	}
+
 	configCopy := config.DeepCopy()
 	// Process triggers and start an initial rollouts
-	shouldTrigger, shouldSkip := triggerActivated(configCopy, latestExists, latestDeployment, c.codec)
-	if !shouldSkip && shouldTrigger {
-		configCopy.Status.LatestVersion++
-		return c.updateStatus(configCopy, existingDeployments)
+	shouldTrigger, shouldSkip, err := triggerActivated(configCopy, latestExists, latestDeployment, c.codec)
+	if err != nil {
+		return fmt.Errorf("triggerActivated failed: %v", err)
+	}
+
+	if shouldSkip {
+		return c.updateStatus(configCopy, existingDeployments, true)
 	}
-	// Have to wait for the image trigger to get the image before proceeding.
-	if shouldSkip && appsutil.IsInitialDeployment(config) {
-		return c.updateStatus(configCopy, existingDeployments)
+
+	if shouldTrigger {
+		configCopy.Status.LatestVersion++
+		_, err := c.dn.DeploymentConfigs(configCopy.Namespace).UpdateStatus(configCopy)
+		return err
 	}
+
 	// If the latest deployment already exists, reconcile existing deployments
 	// and return early.
 	if latestExists {
 		// If the latest deployment is still running, try again later. We don't
 		// want to compete with the deployer.
 		if !appsutil.IsTerminatedDeployment(latestDeployment) {
-			return c.updateStatus(config, existingDeployments)
+			return c.updateStatus(config, existingDeployments, false)
```

Comment:
is this the only case when we don't update the observed generation? maybe it will be nicer to have two funcs: `updateStatusAndGeneration` and `updateStatusWithoutGeneration`... (i really don't like the boolean params as one have to always search what are they for ;-)

Reply:
I don't think it's worth it to split 1 func into 3 (1 internal) just because of the bool; I don't feel strong about it though